packages/@aws-cdk/aws-codebuild/lib/source.ts

```diff
@@ -71,17 +71,48 @@ export class NoSource extends BuildSource {
   }
 }
 
+/**
+ * The construction properties common to all build sources that are backed by Git.
+ */
+export interface GitBuildSourceProps extends BuildSourceProps {
+  /**
+   * The depth of history to download. Minimum value is 0.
+   * If this value is 0, greater than 25, or not provided,
+   * then the full history is downloaded with each build of the project.
+   */
+  cloneDepth?: number;
+}
+
+/**
+ * A common superclass of all build sources that are backed by Git.
+ */
+export abstract class GitBuildSource extends BuildSource {
+  private readonly cloneDepth?: number;
+
+  protected constructor(props: GitBuildSourceProps) {
+    super(props);
+
+    this.cloneDepth = props.cloneDepth;
+  }
+
+  public toSourceJSON(): CfnProject.SourceProperty {
+    const ret = super.toSourceJSON();
+    ret.gitCloneDepth = this.cloneDepth;
+    return ret;
+  }
+}
+
 /**
  * Construction properties for {@link CodeCommitSource}.
  */
-export interface CodeCommitSourceProps extends BuildSourceProps {
+export interface CodeCommitSourceProps extends GitBuildSourceProps {
   repository: codecommit.IRepository;
 }
 
 /**
  * CodeCommit Source definition for a CodeBuild project.
  */
-export class CodeCommitSource extends BuildSource {
+export class CodeCommitSource extends GitBuildSource {
```

Comment:
BTW, based on our discussions around #1743, this class should technically be in CodeCommit...

Reply:
I think you're probably right, but that seems like a separate change.